### PR TITLE
[Merged by Bors] - chore: reduce imports in ZMod.Basic

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
- While at it, document the lemmas that were moved, and two similar lemmas in `ZMod.Basic`.

---
- The extra `import Mathlib.RingTheory.ZMod` import in `Mathlib/Data/ZMod/Quotient.lean` and `Mathlib/NumberTheory/Padics/RingHoms.lean` is probably not too bad (shake didn't speak yet, so perhaps that is even superfluous).

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Commenting to avoid duplicate work: I'm trying to pick this up.

**Comment by grunweg** [comment]
This is *almost* ready: there is one build error to fix in `Data/Zmod/Quotient`.

**Comment by grunweg** [comment]
@Ruben-VandeVelde I'd declare this done (modulo further build errors, but I cannot imagine those). Feel free to take a look.

File: Mathlib/Data/ZMod/Module.lean [hunk 1]
```diff
@@ -3,8 +3,9 @@ Copyright (c) 2023 Lawrence Wu. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Lawrence Wu
 -/
+import Mathlib.Order.OmegaCompletePartialOrder
 import Mathlib.Data.ZMod.Basic
-import Mathlib.Algebra.Module.LinearMap.Basic
+import Mathlib.Algebra.Module.Submodule.Lattice
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/ZMod/Module.lean:8]: While you're here, please sort these

File: Mathlib/RingTheory/ZMod.lean [hunk 2]
```diff
@@ -12,14 +12,32 @@ import Mathlib.RingTheory.PrincipalIdealDomain
...
 # Ring theoretic facts about `ZMod n`
 
-We collect a few facts about `ZMod n` that need some ring theory to be proved/stated
+We collect a few facts about `ZMod n` that need some ring theory to be proved/stated.
 
 ## Main statements
 
-* `isReduced_zmod` - `ZMod n` is reduced for all squarefree `n`.
+* `ZMod.ker_intCastRingHom`:  the ring homomorphism `ℤ → ZMod n` has kernel generated by `n`.
+* `ZMod.ringHom_eq_of_ker_eq`: zwo ring homomorphisms into `ZMod n` with equal kernels are equal.
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/ZMod.lean:20]:
```suggestion
* `ZMod.ringHom_eq_of_ker_eq`: two ring homomorphisms into `ZMod n` with equal kernels are equal.
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
The parts I didn't do look good :)

File: Mathlib/Data/ZMod/Basic.lean [hunk 3]
```diff
@@ -4,9 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes
 -/
 import Mathlib.Algebra.CharP.Basic
-import Mathlib.RingTheory.Ideal.Operations
-import Mathlib.Data.Fintype.Units
-import Mathlib.Data.Nat.Parity
+import Mathlib.Algebra.Ring.Prod
 import Mathlib.Tactic.FinCases
```
Comment (grunweg) [inline Mathlib/Data/ZMod/Basic.lean:8]: I think this can be removed, #12487 confirms this.
```suggestion

Comment (Ruben-VandeVelde) [reply Mathlib/Data/ZMod/Basic.lean:8]: Presumably because it's imported transitively, because this file does use `fin_cases`

**Review by Vierkantor — APPROVED** [review]
I also complained about the fact that `shake` doesn't warn about transitive imports, but apparently that's supposed to be a feature. So I'll let you decide whether it's best to keep or remove.

Looks good to me otherwise, thanks!

bors d+

**Comment by Ruben-VandeVelde** [comment]
bors merge

**Comment by grunweg** [comment]
@Ruben-VandeVelde This builds again :-)

**Comment by Ruben-VandeVelde** [comment]
bors merge

**Pull request closed without merging**